packages/api/src/derives/fees/estimateFee.ts

```diff
@@ -0,0 +1,61 @@
+import {ApiInterfaceRx} from '@cennznet/api/types';
+import {DEFAULT_FEE_ASSET} from '@cennznet/types/extrinsic/constants';
+import Extrinsic from '@cennznet/types/extrinsic/Extrinsic';
+import {ChargeTransactionPayment} from '@cennznet/types/runtime/transaction-payment';
+import {AnyAssetId, IExtrinsic} from '@cennznet/types/types';
+import {drr} from '@polkadot/rpc-core/rxjs';
+import {TypeRegistry} from '@polkadot/types';
+import {Address} from '@polkadot/types/interfaces';
+import {RuntimeDispatchInfo} from '@polkadot/types/interfaces/rpc';
+import ExtrinsicEra from '@polkadot/types/primitive/Extrinsic/ExtrinsicEra';
+import BN from 'bn.js';
+import {combineLatest, Observable, of} from 'rxjs';
+import {catchError, first, map, switchMap} from 'rxjs/operators';
+
+export function estimateFee(api: ApiInterfaceRx) {
+  return (
+    extrinsic: IExtrinsic,
+    feeAssetId: AnyAssetId,
+    sender: Address,
+    transactionPayment: ChargeTransactionPayment
```

Comment:
Did you try faking `transactionPayment` with some high number?

Reply:
yes.. it works well provided the maxPayment is same while estimating and while send an extrinsic.